src/Microsoft.VisualStudio.Services.Agent/WebProxy.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Common;
+using System;
+using System.IO;
+using System.Net;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public class WebProxy : IWebProxy
+    {
+        public WebProxy(Uri proxyAddress)
+        {
+            if (proxyAddress == null)
+            {
+                throw new ArgumentNullException(nameof(proxyAddress));
+            }
+
+            ProxyAddress = proxyAddress;
+        }
+
+        public Uri ProxyAddress { get; private set; }
+
+        public ICredentials Credentials { get; set; }
+
+        public Uri GetProxy(Uri destination) => ProxyAddress;
+
+        public bool IsBypassed(Uri uri)
+        { 
+            return false;
+        }
+
+        private static bool _proxySettingsApplied = false;
+
+        public static void ApplyProxySettings()
+        {
```

Comment:
This looks susceptible to a race condition. Although I don't know whether it's feasible based on the caller logic. The double-check-and-lock pattern would eliminate the concern:
```
if (!_proxySettingsApplied)
{
    lock (_lock)
    {
        if (!Volative.Read(_proxySettingsApplied))
        {
            try
            {
                // logic to set DefaultWebProxy
            }
            finally
            {
                _proxyConfigured = true;
            }
        }
    }
}